page-optimize.php

```diff
@@ -21,6 +21,47 @@
 	exit;
 }
 
+if( ! wp_next_scheduled( 'page_optimize_cache_cleanup' ) ) {
+	wp_schedule_event( time(), 'daily', 'page_optimize_cache_cleanup' );
+}
+
+function page_optimize_cache_cleanup() {
+	if ( ! is_dir( PAGE_OPTIMIZE_CACHE_DIR ) ) {
+		return;
+	}
+
+	// Grab all files in the cache directory
+	$cache_files = array_filter(
+		scandir( PAGE_OPTIMIZE_CACHE_DIR ),
+		function ( $file ) {
+			return is_file( PAGE_OPTIMIZE_CACHE_DIR . "/$file" );
+		}
+	);
+
+	// Cleanup all files older than 24 hours
+	foreach ( $cache_files as $cache_file ) {
+		$cache_file_full_path = PAGE_OPTIMIZE_CACHE_DIR . "/$cache_file";
+		if ( ( time() - DAY_IN_SECONDS ) > filemtime( $cache_file_full_path ) ) {
+			unlink( $cache_file_full_path );
+		}
+	}
+}
+
+// Unschedule cache cleanup, and purge cache directory
+function page_optimize_deactivate() {
+	// Purge cache dir if it exists
+	if ( is_dir( PAGE_OPTIMIZE_CACHE_DIR ) ) {
+		array_map(
+			'unlink',
+			glob( PAGE_OPTIMIZE_CACHE_DIR . '/*.*' )
```

Comment:
What happens when PAGE_OPTIMIZE_CACHE_DIR is something like `/srv/htdocs`? The cache would work while the plugin was active, but on deactivation, we'd remove everything that matches `/srv/htdocs/*.*`.

Maybe we can give the cache files some hopefully-unique prefix so we can match prefixed things only.